pkg/asset/machines/openstack/worker.go

```diff
@@ -47,8 +47,7 @@ spec:
         value:
           apiVersion: openstack.cluster.k8s.io/v1alpha1
           kind: OpenStackMachineProviderConfig
-          image:
-            id: {{.Image}}
+          image: {{.Image}}
```

Comment:
We also need all of these (using my own hardcoded values):

```yaml
      cloudName: standalone
      cloudsSecret: openstack-creds
      networks:
      - uuid: c24f8bc2-4d8b-4ab3-a6ed-1be816113bf6
      placement:
        region: regionOne
      securityGroups:
      - worker
      subnet: worker
```

And basically the same thing for the master template.

Reply:
I knew I was forgetting something. Will update

Reply:
From your comment, I'm guessing that using tags didn't work for you. Right?

Reply:
In what way? If you mean for worker deletion, than no they didn't.

Btw thanks for the update. As far as I can see, it's still missing the `networks` array and fixing up the `subnet` name and `securityGroups`. Will you add those as well please? I've had to fix all three of them in my testing.

Reply:
@tomassedovic sorry, I meant for filtering subnets, networks, and securityGroups. The keys exist in the template but I'm assuming that the tag filter didn't work for you.

Reply:
@flaper87 ah yes, sorry.

You're correct: the filters did not work. The actuator kept complaining that it could not be deserialised properly and that it expected a string (or an array of strings for SG).

As far as I can see, this is the validation it uses:

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/config/crd/openstackproviderconfig_v1alpha1.yaml#L19-L28

(or at least something consistent with those expectations)

Reply:
@tomassedovic right, that's what I thought. I think this should be fixed in the actuator CRD since we've added support for filters already: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/openstack/clients/machineservice.go#L255

What do you think about getting this patch in as is and check these filters one by one in the actuator?